tracer/src/Datadog.Trace/Configuration/ConfigurationSources/Telemetry/ConfigurationBuilder.cs

```diff
@@ -156,80 +100,156 @@ public string AsRedactedString(string defaultValue)
         /// If you can, use <see cref="AsString(string, Func&lt;string, bool&gt;?)" /> instead or record telemetry manually.
         /// </summary>
         /// <returns>the string value of the configuration if set and valid</returns>
-        public string? AsString(Func<string, bool> validator) => AsString(getDefaultValue: null, validator, recordValue: true);
+        public string? AsString(Func<string, bool> validator) => AsString(defaultValue: null, validator, recordValue: true);
 
         public string AsString(string defaultValue, Func<string, bool>? validator)
-            => AsString(() => defaultValue, validator, recordValue: true);
+            => AsString(defaultValue, validator, recordValue: true);
 
         [return: NotNullIfNotNull(nameof(getDefaultValue))]
-        public string? AsString(Func<DefaultResult<string>>? getDefaultValue, Func<string, bool>? validator)
+        public string? AsString(Func<string>? getDefaultValue, Func<string, bool>? validator)
             => AsString(getDefaultValue, validator, recordValue: true);
 
         [return: NotNullIfNotNull(nameof(getDefaultValue))]
-        public string? AsString(Func<DefaultResult<string>>? getDefaultValue, Func<string, bool>? validator, Func<string, ParsingResult<string>> converter)
+        public string? AsString(Func<string>? getDefaultValue, Func<string, bool>? validator, Func<string, ParsingResult<string>> converter)
             => AsString(getDefaultValue, validator, converter, recordValue: true);
 
         [return: NotNullIfNotNull(nameof(getDefaultValue))]
-        private string? AsString(Func<DefaultResult<string>>? getDefaultValue, Func<string, bool>? validator, bool recordValue)
+        private string? AsString(Func<string>? getDefaultValue, Func<string, bool>? validator, bool recordValue)
             => AsString(getDefaultValue, validator, converter: null, recordValue);
 
+        [return: NotNullIfNotNull(nameof(defaultValue))]
+        private string? AsString(string? defaultValue, Func<string, bool>? validator, bool recordValue)
```

Comment:
The difference between these two methods is the first is `T? defaultValue` and the second is `Func<T>? defaultValue`. We should favour the former where possible